app/models/user.rb

```diff
@@ -32,12 +31,47 @@ def session_token
     remember_digest || remember
   end
 
-  def authenticated?(remember_token) 
-    return false if remember_digest.nil?
-    BCrypt::Password.new(remember_digest).is_password?(remember_token)
-  end 
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil? 
```

Comment:
It's recommended that you leave an empty line after a guard clause like this one